pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -1798,11 +1798,12 @@ public ResultSet getTablePrivileges(@Nullable String catalog, @Nullable String s
     f[6] = new Field("IS_GRANTABLE", Oid.VARCHAR);
 
     String sql;
+    // r = ordinary table, p = permanent table, v = view, m = materialized view, f = foreign table
```

Comment:
"p" is a partitioned table.

See "relkind" in the pg_class docs: https://www.postgresql.org/docs/current/catalog-pg-class.html

Reply:
Thank you, got confused with relpersistence